### PR TITLE
ConDec-498: Manage file system directories for branches of git repositories

### PR DESCRIPTION
**Pull request by lw2011**
Here is the git repository file system manager our gitClient implementation should use in the future for multi-user/multi-request operation of our plugin.
After this branch the gitClientImpl should be adjusted or an alternative implementation could be used.

### CODE REVIEW

File: src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/GitRepositoryFSManager.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Date;
+
+public class GitRepositoryFSManager {
+	private static final String TEMP_DIR_PREFIX = "TEMP";
+	private static final long BRANCH_OUTDATED_AFTER = 60 * 60 * 1000; //ex. 1 day = 24 hours * 60 minutes * 60 seconds * 1000 miliseconds
+	private String basePath;
```
[the remark this answers is not in the document]
Comment (kleebaum) [reply src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/GitRepositoryFSManager.java]: @lw2011 , thanks for the pull request! Could you please have a look at the issues found by Codacy and fix them if you think they are reasonable?

File: src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/GitRepositoryFSManager.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+		}
+		else {
+			Date date= new Date();
+			long time = date.getTime();
+			String tempDirString = baseProjectUriPath
+					+File.separator
+					+TEMP_DIR_PREFIX+String.valueOf(time);
+			File tempDir = new File(tempDirString);
+			try {
+                /* TODO: An issue is observable on Windows machine.
```
Comment (kleebaum) [inline src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/GitRepositoryFSManager.java]: An issue is "observed"? You might indicate the issue by the @issue tag in the comment.

File: src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/GitRepositoryFSManager.java [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+					" nor default folder could be found under: "
+					+ baseProjectUriPath);
+			return null;
+		}
+		rememberBranchPathRequest(branchShortName);
+		return getBranchPath(branchShortName);
+	}
+
+	/*
+	 * @issue:file system does not allow all charcters for folder and file name,
```
Comment (kleebaum) [inline src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/GitRepositoryFSManager.java]: Great that you use the knowledge tags here!

**Comment by lw2011** [comment]
I rebased this branch to most current develop branch and pushed a commit with fixable codacy issues.

**Pull request merged**